src/privateStore.js

```diff
@@ -65,30 +72,58 @@ class PrivateStore {
    * @param     {String}    hash                        The hash of the private store OrbitDB
    */
   async _sync (hash) {
-    // if hash === null; create a new orbitdb instance.
-    const orbitdb = new OrbitDB(this.ipfs)
-    //TODO - sync the OrbitDB key-value store
+    if (!this.db) {
+      const orbitdb = new OrbitDB(this.ipfs)
+      // the db needs a unique name, we use the hash of the DID + a store specific name
+      const storeName = multihash(this.muportDID.getDid()) + ".datastore"
+      this.db = await orbitdb.keyvalue(storeName, {
+        replicate: false,
+        write: ['*']
+      })
+    }
+    if (hash) {
+      // sync orbitdb to hash
+      let log = await Log.fromEntryHash(this.ipfs, hash)
+
+      return new Promise((resolve, reject) => {
+        this.db.events.on('replicated', async (address, logLength) => {
+          // get the key salt of the db
+          const encryptedSalt = await this.db.get(SALT_KEY)
+          this.salt = this._decryptEntry(encryptedSalt)
+          resolve()
+        })
+        this.db.sync(log.values)
+      })
+    }
+    // This is the first time the store is used.
+    // Generate a random salt and save in the db.
+    this.salt = Buffer.from(nacl.randomBytes(16)).toString('hex')
+    const encryptedSalt = this._encryptEntry(this.salt)
+    await this.db.put(SALT_KEY, encryptedSalt)
+  }
+
+  async close () {
+    this.db.close()
   }
 
   _genDbKey (key) {
-    // return someHashFunction(key + this.salt)
+    return multihash(this.salt + key)
   }
 
   _encryptEntry (entry) {
-    // TODO - use muport to encrypt entry
+    const encrypted = this.muportDID.symEncrypt(entry)
+    return encrypted.nonce + '.' + encrypted.ciphertext
   }
 
   _decryptEntry (entry) {
-    // TODO - use muport to decrypt entry
+    let [nonce, ciphertext] = entry.split('.')
+    return this.muportDID.symDecrypt(ciphertext, nonce)
   }
 }
 
-// some sample code for  creating a log store
-//orbitdb.log('hello', {replicate: false}).then(db => {
-  //db.add('asdfasdf').then(h => {
-    //const eve = db.get(h)
-    //console.log(eve)
-  //})
-//}).catch(console.log)
+const multihash = str => {
+  const dataBuf = Buffer.from(str, 'utf8')
+  return Multihash.encode(dataBuf, 'sha3-256').toString('hex')
```

Comment:
Why the use of `sha3-256` here? Why not `sha2-256` (more standard) or `keccak256` (more ethereum-y)?

Reply:
Tnx, will update